### ISSUE TITLE
linter: `extends` should resolve the path to workspace package

### ISSUE THREAD

**Issue by Miodec** [oxc-project/oxc#10395]
### What version of Oxlint are you using?

0.16.6

### What command did you run?

both cli and vsc extension

### What does your `.oxlintrc.json` config file look like?

```jsonc
{
  "extends": [
    // "../packages/oxlint-config/index.json",
    "@monkeytype/oxlint-config"
  ],
  "overrides": [
    {
      "files": ["__tests__/**"],
      "plugins": ["jest", "vitest"],
      "rules": {
        // "no-unused-vars": "off"
      }
    }
  ]
}

```

### What happened?

When trying to use monorepo local workspace packages, the path is not resolved correctly (in ESLint this works fine).

This doesn't work - when switching these around, everything works as expected.
```jsonc
  "extends": [
    // "../packages/oxlint-config/index.json",
    "@monkeytype/oxlint-config"
  ],
```

Verified its installed  as a dev dependency:
![Image](https://github.com/user-attachments/assets/bede227f-bc29-4201-81ee-d8a1fc265b18)

The file is exported (yes, its called index.json)
![Image](https://github.com/user-attachments/assets/639a3211-d903-40b6-a198-c02a6b33f694)

Not working 
![Image](https://github.com/user-attachments/assets/3c60341d-f8ef-4981-81bc-13c16d68cbb2)

Working (unused var is now an error)
![Image](https://github.com/user-attachments/assets/f4d104be-06ce-4d9b-b951-28135ce9325f)

**Comment by camchenry** [comment]
This is explicit behavior for the moment, we decided not to add any resolution to the config file paths initially because it is faster and simpler. For now, you can specify the file path explicitly. I'll leave this issue open for discussion though.

**Comment by Miodec** [comment]
> This is explicit behavior for the moment, we decided not to add any resolution to the config file paths initially because it is faster and simpler. For now, you can specify the file path explicitly. I'll leave this issue open for discussion though.

It would be a nice-to-have to support both fast relative path and slower resolved paths. In large mono-repos if the base config ever changes path, you would need to update `.oxlintrc.json` in every package.

Unless the performance hit would be felt on every run, then i guess im fine with that.

**Comment by krispya** [comment]
It's much easier to create a template package when the configs are also packages and the paths resolved. I would love to see this!

**Comment by krispya** [comment]
Also, would plugins require this anyways? Since the JSON would need to resolve the path to the installed package? https://github.com/oxc-project/oxc/issues/9905

**Comment by ConsoleTVs** [comment]
In our case, we have a shared standards package where we export common base configurations for stuff like `eslint`, `prettier`, `tsconfig` among others. I wanted to add the `oxlint` config too but ran into the same issue. We can't extend a package.

I think we should either allow the JSON to resolve paths or allow programatic configurations (`.ts`/`.js` files) so we can resolve this ourselves.

**Comment by krispya** [comment]
I am doing this too. As an ugly workaround you can have the path go directly to `node_modules`.

```
{
	"extends": ["node_modules/@config/oxlint/base.json"]
}
```

**Comment by stefanprobst** [comment]
> This is explicit behavior for the moment,

will this be reconsidered @camchenry ?

**Comment by camchenry** [comment]
This is effectively solved already, but the documentation here doesn't represent it as well as it could. I've opened https://github.com/oxc-project/website/issues/1093 to address that.

In short: use `oxlint.config.ts` for this usecase:

```ts
// oxlint.config.ts
import sharedConfig from "@example-org/oxlint-config";
import { defineConfig } from "oxlint";

export default defineConfig({
  extends: [sharedConfig],
});
```
